### DISCUSSION TITLE
New particle source for simulation of CT-scanner

### DISCUSSION THREAD

**Discussion by malukuhlmann** [nrc-cnrc/EGSnrc#1200] [Ideas]
Hi,

I would like to present a new particle source I have implemented and validated (see [Kuhlmann, 2024](https://iopscience.iop.org/article/10.1088/1361-6560/ad3886)). 
Is there a possibility to have it included in EGSnrc? 

I have added a description below. 

The edited files can be found under: https://gitlab1.ptb.de/mkuhlmann/egsnrc_with_aleqctsource/-/tree/al_eq_ct_source?ref_type=heads

An example for usage can be found under HEN_HOUSE/user_codes/cavity

Kind regards,
Malu

# al_eq_ct_source

## Description
A source that simulates the photon fluence of an aluminum equivalent computer tomography source model. The 
model can be adapted to any CT scanner by measuring the aluminum equivalent spectrum and then calculating the bowtie filter's aluminum equivalent thickness. Measurement and calculation procedures are described in the following publications: 

[Ahlikani, 2016](https://www.sciencedirect.com/science/article/abs/pii/S1120179715009060?via%3Dihub),
[Rosendahl, 2019](https://iopscience.iop.org/article/10.1088/1361-6560/ab0e97)

The implemented source was benchmarked against measurements and an uncertainty analysis was carried out. The results are 
presented in this publication:

[Kuhlmann, 2024](https://iopscience.iop.org/article/10.1088/1361-6560/ad3886)

![source_swb](https://github.com/user-attachments/assets/f6dd3b4b-f774-40e4-b761-a9d4c5e6db78)

Fig.1: A simple example of fluence output with geometrical annotations.

**Comment by rtownson** [comment]
Looks really interesting! Well done designing the source properly for egs++. You are welcome to submit it as a pull request, or if you're not sure how to do that we can give you a hand. We'll be very busy for the next few weeks, but hopefully can take a closer look after that.

**Reply by malukuhlmann** [reply]
Is there anywhere a instructions how to make a pull request? 
I already created my own branch on my computer (it contains all the changes from the develop branch and my own changes). How can I now push it? I read something about an accesstoken? Where can I create this? 
Thank you!

**Reply by rtownson** [reply]
If you want to avoid the hassle, you are always welcome to just email the changed files in a zip file. You'll still get credit in file headers and the pull request, but you won't get the "number of lines touched" record in our stats.

To do it through github, you would first create a "fork" of EGSnrc on your own github account. 

The maybe easier way:  Clone that fork onto your computer. Create a new branch on that fork and copy your changed files into it. Commit, push, and then on github there's an option to create a pull request from your branch. You will have to set up ssh access tokens for your account to be able to push.

The better way: If you have a long commit history that there's value in preserving (insightful commit messages, for example), then add your github fork as a remote to your current working repository (instead of cloning a new repo), using `git remote add ...etc...`. Then you'll be able to push your new branch there, after adding ssh tokens, and then create a PR.

If you get stuck anywhere just let me know :)

**Reply by rtownson** [reply]
For push access to your own fork:

Make sure you clone with ssh (not https).

See the `Generating a new SSH key` section here: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

Then you would copy the contents of the .pub file it creates into your github ssh keys (under settings).
